domains/certificates/Query.API/API/ContractService/Clients/MeteringPointsClient.cs

```diff
@@ -33,22 +32,45 @@ public MeteringPointsClient(HttpClient httpClient, IHttpContextAccessor httpCont
 
     public async Task<MeteringPointsResponse?> GetMeteringPoints(string owner, CancellationToken cancellationToken)
     {
+        ValidateHttpContext();
         SetAuthorizationHeader();
-
         ValidateOwnerAndSubjectMatch(owner);
+        SetApiVersionHeader();
+
+        var meteringPointsUrl = IsBearerTokenIssuedByB2C() ? $"/api/measurements/meteringpoints?orgId={owner}" : "/api/measurements/meteringpoints";
 
-        return await httpClient.GetFromJsonAsync<MeteringPointsResponse>("/api/measurements/meteringpoints",
-            cancellationToken: cancellationToken, options: jsonSerializerOptions);
+        return await httpClient.GetFromJsonAsync<MeteringPointsResponse>(meteringPointsUrl, jsonSerializerOptions,
+            cancellationToken);
     }
 
     private void SetAuthorizationHeader()
+    {
+        httpClient.DefaultRequestHeaders.Authorization = AuthenticationHeaderValue.Parse(httpContextAccessor.HttpContext!.Request.Headers.Authorization!);
+    }
+
+    private void SetApiVersionHeader()
+    {
+        var downstreamApiVersion = GetDownstreamApiVersion();
+        httpClient.DefaultRequestHeaders.Remove("EO_API_VERSION");
```

Comment:
Why remove and add back?

Reply:
The add method adds an additional header with no check if a header with same name is already added. So we get requests with multiple EO_API_VERSION headers.